README.md

```diff
@@ -70,11 +70,21 @@ Please do not use the issue tracker for security issues.
 
 ## Setup
 
+Ensure you have Node and Yarn installed.
+
+The latest Node version officially supported by OSx and Hardhat is 16. Node >=19 also works, but is technically unsupported by Hardhat. Use [nvm](https://github.com/nvm-sh/nvm) to switch:
+
+```sh
+nvm install 16
+nvm use 16
+npm i -g yarn
+```
```

Comment:
I am wondering if we should point them to the official yarn docs so they can install yarn depending on there OS, instead of this